src/classes/SftpSessionHandler.ts

```diff
@@ -0,0 +1,388 @@
+import fs from 'fs';
+import path from 'path';
+import { v4 as uuidv4 } from 'uuid';
+import { getArchives } from '@permanentorg/sdk';
+import ssh2 from 'ssh2';
+import { logger } from '../logger';
+import type {
+  Attributes,
+  FileEntry,
+  SFTPWrapper,
+} from 'ssh2';
+import type { Archive } from '@permanentorg/sdk';
+
+const getFileType = (filePath: string): number => {
+  // TODO: this is not a real rule -- directories can have extensions and files don't have to have them.
```

Comment:
This will end up being a lookup based on the permanent API's return results once we're loading real data.  For now we have to spoof it like this.